code/controllers/subsystems/timer.dm

```diff
@@ -409,18 +409,50 @@ var/datum/controller/subsystem/timer/SStimer
 /proc/deltimer(id)
 	if (!id)
 		return FALSE
+
 	if (id == TIMER_ID_NULL)
 		CRASH("Tried to delete a null timerid. Use the TIMER_STOPPABLE flag.")
+
 	if (!istext(id))
 		if (istype(id, /datum/timedevent))
 			qdel(id)
 			return TRUE
+
 	var/datum/timedevent/timer = SStimer.timer_id_dict["timerid[id]"]
 	if (timer && !timer.spent)
 		qdel(timer)
 		return TRUE
 	return FALSE
 
+/proc/execute_and_deltimer(id)
+	if (!id)
+		return FALSE
+
+	if (id == TIMER_ID_NULL)
+		CRASH("Tried to delete a null timerid. Use the TIMER_STOPPABLE flag.")
+
+	if (!istext(id))
+		if (istype(id, /datum/timedevent))
+			var/datum/timedevent/timer = id
+			var/datum/callback/callBack = timer.callBack
+			callBack.InvokeAsync()
+			timer.spent = TRUE
+			qdel(id)
+			return TRUE
+
+	var/datum/timedevent/timer = SStimer.timer_id_dict["timerid[id]"]
+
+	if(!timer)
+		return FALSE
+
+	var/datum/callback/callBack = timer.callBack
+
+	if (timer && !timer.spent)
+		callBack.InvokeAsync()
```

Comment:
PSure I already mentioned this. InvokingAsync by default is bad, and _only_ invoking async is worse. Normal code is expected to be synchronous. Make it a parameter as to which method to use `invoke_async = FALSE`.